src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -668,12 +670,29 @@ private void TakeIncrementalSnapshot(BoundNode node)
             _snapshotBuilderOpt?.TakeIncrementalSnapshot(node, State);
         }
 
-        private void SetUpdatedSymbol(BoundNode node, Symbol originalSymbol, Symbol updatedSymbol)
+        private void SetLocalType(BoundNode sourceNode, LocalSymbol originalLocal, TypeWithAnnotations updatedType)
+        {
+            _variableTypes[originalLocal] = updatedType;
+
+            LocalSymbol updatedLocal;
+            if (_reinferredVariablesOpt is object)
+            {
+                updatedLocal = ((SourceLocalSymbol)originalLocal).WithContainingAndAnalyzedTypes(GetUpdatedContainingSymbol(originalLocal), updatedType);
+                if ((object)originalLocal == updatedLocal) return;
```

Comment:
>return [](start = 59, length = 6)

Do we need to call `SetUpdatedSymbol()` always, even if the value is the original, in case the map was updated previously with a distinct value?

Reply:
No longer relevant.

---
In reply to: [327870685](https://github.com/dotnet/roslyn/pull/38769#discussion_r327870685) [](ancestors = 327870685)